File: api/src/main/java/com/aistore/module/itemimage/entity/ItemImage.java
package com.aistore.module.itemimage.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/** 物料图片实体（表 item_image，spu/sku/level 三者可空） */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("item_image")
public class ItemImage {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;
    private Long spuId;
    private Long skuId;
    private Long levelId;
    private String imageUrl;
    private String imageType;
    private Integer sortOrder;
    private Integer isPrimary;
    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    private Long createdBy;
    private Long updatedBy;

    @TableLogic
    private Integer deleted;
}

File: web/src/types/itemImage.ts
/* 物料图片模块类型（对齐 OpenAPI 契约） */
export interface ItemImage {
  id: number;
  spuId?: number | null;
  skuId?: number | null;
  levelId?: number | null;
  imageUrl: string;
  imageType?: string | null;
  sortOrder: number;
  isPrimary: 0 | 1;
  status: 0 | 1;
  createdAt: string;
  updatedAt: string;
  createdBy?: number | null;
  updatedBy?: number | null;
}

export interface ItemImageSummary {
  id: number;
  spuId?: number | null;
  skuId?: number | null;
  levelId?: number | null;
  imageUrl: string;
  imageType?: string | null;
  sortOrder: number;
  isPrimary: 0 | 1;
  status: 0 | 1;
  createdAt: string;
}

export interface ItemImageListResponse {
  items: ItemImageSummary[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface CreateItemImageRequest {
  spuId?: number;
  skuId?: number;
  levelId?: number;
  imageUrl: string;
  imageType?: string;
  sortOrder?: number;
  isPrimary?: 0 | 1;
  status?: 0 | 1;
}

export interface UpdateItemImageRequest {
  imageUrl?: string;
  imageType?: string;
  sortOrder?: number;
  isPrimary?: 0 | 1;
}

export interface ItemImageQueryParams {
  spuId?: number;
  skuId?: number;
  levelId?: number;
  status?: 0 | 1;
  page?: number;
  pageSize?: number;
}
